### DISCUSSION TITLE
solar radiation

### DISCUSSION THREAD

**Discussion by wiesehahn** [r-lidar/lidR#598] [General]
I would like to know if someone tried to or knows how to calculate solar radiation from point clouds using lidR (maybe just for some steps)? 

I have tried [timeindaylight](https://www.whiteboxgeo.com/manual/wbt_book/available_tools/geomorphometric_analysis.html#TimeInDaylight) from Whitebox Tools in R (applied on a dsm created with lidR). However this is just one step which is not yet the actual radiation. And also it is applied to the surface but not possible to apply for e.g. the understory.

I could imagine different results when some model is applied directly on the point cloud. 

Outside R I came across [https://github.com/3dgeo-heidelberg/vostok](https://github.com/3dgeo-heidelberg/vostok) which i have not tried yet.

**Comment by Jean-Romain** [comment]
I don't know what you are looking for. Is it something like 

- https://twitter.com/piotrtompalski/status/1258820627458191360


or very similar

- https://www.waldrn.com/the-shadiest-spots-in-downtown-indianapolis/

**Reply by wiesehahn** [reply]
In your examples shadows are calculated based on a digital surface model and sun positions. If you sum up these shadows for hourly sun positions throughout the entire year you get something like the inverse of sun hours for the surface. This is similar to what I did with Whitebox Tools time_in_daylight. 
For the surface (e.g. upper canopy) I think this is very much related to solar radiation. Although solar radiation is also dependent on e.g. diffuse radiation effected by cloud cover, surrounding etc.
But this approach models the incoming light only for the surface. 
What I am looking for is a way to estimate the radiation also below the upper canopy and in canopy gaps. 
This could be modeled indirectly as e.g. in [Estimation of sub-canopy solar radiation from LiDAR discrete returns in mixed temporal forest of Białowieża, Poland](https://doi.org/10.1016/j.jag.2019.03.005). 
However, I was wondering if there is a way to directly estimate it from the point cloud. To my understanding this is what the VOSTOK software aims at by assuming that each return point occupies a certain voxel which blocks or reflects light.

**Reply by Jean-Romain** [reply]
Maybe you should look like to DART https://dart.omp.eu

**Reply by ghost** [reply]
@wiesehahn did you find a solution that worked for you? I am interested in the same thing.

**Reply by wiesehahn** [reply]
I am still interested in the topic but it is just a side project and so far I did not have time to dive deeper into it. 

If I find the time and come across something I will mention it here. 
I would be thankful if you could do too.

**Reply by ghost** [reply]
I am talking to people to see how one might go about it to implement it in DART.

In the meantime, I found this https://github.com/fwestling/PruneTreeLS (Paper: https://www.sciencedirect.com/science/article/pii/S016816992100291X?via%3Dihub). I'm going to give it a go and see if it is suitable.

**Reply by wiesehahn** [reply]
referencing https://github.com/r-lidar/lidR/discussions/809
